src/ui/QuerySuggestPreview/QuerySuggestPreview.ts

```diff
@@ -1,59 +1,272 @@
-import { ResultList } from '../ResultList/ResultList';
-import { IResultListOptions } from '../ResultList/ResultListOptions';
 import { IComponentBindings } from '../Base/ComponentBindings';
 import { exportGlobally } from '../../GlobalExports';
-import { ComponentOptions, OmniboxEvents, Initialization } from '../../Core';
+import { ComponentOptions, OmniboxEvents, Initialization, $$, Component } from '../../Core';
 import { IQuerySuggestSelection } from '../../events/OmniboxEvents';
+import { IQueryResults } from '../../rest/QueryResults';
+import 'styling/_QuerySuggestPreview';
+import { ResultListRenderer } from '../ResultList/ResultListRenderer';
+import { IInitializationParameters } from '../Base/Initialization';
+import { IResultListOptions } from '../ResultList/ResultListOptions';
+import { Template } from '../Templates/Template';
+import { TemplateComponentOptions } from '../Base/TemplateComponentOptions';
+import { ResultListTableRenderer } from '../ResultList/ResultListTableRenderer';
+import { ITemplateToHtml, TemplateToHtml } from '../Templates/TemplateToHtml';
+import { IQueryResult } from '../../rest/QueryResult';
 
-export interface IQuerySuggestPreview extends IResultListOptions {
+export interface IQuerySuggestPreview {
   numberOfPreviewResults?: number;
-  width?: string;
+  previewWidth?: number;
+  suggestionWidth?: string;
+  resultTemplate?: Template;
+  headerText?: string;
+  executeQueryDelay?: number;
 }
 
-export class QuerySuggestPreview extends ResultList implements IComponentBindings {
+/**
+ * The `QuerySuggestPreview` component allows you to preview the top query result items of hovered query suggestions.
```

Comment:
Yes it only does work with Coveo ML query suggestions

Yes we can use all the result template component
We should be able to use multiple result template, I did not try it yet so I will confirm it to you later. 
Same for styling for class

Reply:
Then please update the code example in the description as follows:

```html
 <div class="CoveoQuerySuggestPreview">
   <script class="result-template" type="text/html">
     <div class="coveo-result-frame">
       <a class="CoveoResultLink"></a>
     </div>
   </script>
 </div>
```

Also instead of:

> this component requires a [result template](https://docs.coveo.com/en/413/) in its markup configuration to be able to render previews.

I think we should say:

> this component requires at least one [result template](https://docs.coveo.com/en/413/) in its markup configuration to be able to render previews.